pygti/schemas.py

```diff
@@ -84,7 +95,74 @@
         "filter": [FilterEntry],
         "serviceTypes": [FilterServiceType],
         "useRealtime": bool,
+    },
+)
+
+SIRequest = Schema.extend(BaseRequestType, {"station": SDName})
+
+ContSearchByServiceId = Schema(
+    {
+        "serviceId": int,
+        "lineKey": str,
+        "plannedDepArrTime": GTITime,
+        "additionalOffset": int,
     }
 )
 
-SIRequest = Schema({"station": SDName})
+# kinds: GTI Docs say int but traffic on the website uses strings
+# Same thing with the value of the penalty. All are strings
+TariffInfoSelector = Schema({"tariff": str, "tariffRegions": bool, "kinds": [str],})
+PenaltyName = In(
+    [
+        "ChangeEvent",
+        "ExtraFare",
+        "Walker",
+        "AnyHandicap",
+        "ToStartStationBy",
+        "TimeRange",
+        "ForVisitors",
+        "desiredType",
+        "DesiredLine",
+    ]
+)
+
+Penalty = Schema({"name": PenaltyName, "value": str})
+
+RealtimeType = In(["REALTIME", "PLANDATA"])
+
+SimpleServiceType = In(
+    ["BUS", "TRAIN", "SHIP", "FOOTPATH", "BICYCLE", "AIRPLANE", "CHANGE"]
```

Comment:
The WADL also lists "CHANGE_SAME_PLATFORM"

Reply:
can't find anything like that. Can you send me a link to the wadl?

Reply:
https://api-test.geofox.de/gti/public/geofoxThinInterfacePublic.xsd